tests/unittests/classification/test_group_fairness.py

```diff
@@ -0,0 +1,127 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+
+import numpy as np
+import pytest
+import torch
+from fairlearn.metrics import demographic_parity_ratio, true_positive_rate_ratio
+from scipy.special import expit as sigmoid
+
+from torchmetrics.classification.group_fairness import BinaryFairness
+from torchmetrics.functional.classification.group_fairness import binary_fairness
+from unittests.classification.inputs import _group_cases
+from unittests.helpers import seed_all
+from unittests.helpers.testers import THRESHOLD, MetricTester, inject_ignore_index, remove_ignore_index_groups
+
+seed_all(42)
+
+
+def _fairlearn_binary(preds, target, groups, ignore_index):
+    preds = preds.numpy()
+    target = target.numpy()
+    groups = groups.numpy()
+
+    if np.issubdtype(preds.dtype, np.floating):
+        if not ((0 < preds) & (preds < 1)).all():
+            preds = sigmoid(preds)
+        preds = (preds >= THRESHOLD).astype(np.uint8)
+
+    target, preds, groups = remove_ignore_index_groups(target, preds, groups, ignore_index)
+
+    dp = demographic_parity_ratio(target, preds, sensitive_features=groups)
+    eo = true_positive_rate_ratio(target, preds, sensitive_features=groups)
+
+    # need to create a Dict here
+    return dp, eo
+
+
+@pytest.mark.parametrize("input", _group_cases)
+class TestBinaryFairness(MetricTester):
+    @pytest.mark.parametrize("ignore_index", [None, 0, -1])
+    @pytest.mark.parametrize("ddp", [False, True])
+    def test_binary_fairness(self, ddp, input, ignore_index):
+        preds, target, groups = input
+        if ignore_index == -1:
+            target = inject_ignore_index(target, ignore_index)
+
+        self.run_class_metric_test(
+            ddp=ddp,
+            preds=preds,
+            target=target,
+            metric_class=BinaryFairness,
+            reference_metric=partial(_fairlearn_binary, ignore_index=ignore_index),
+            metric_args={"threshold": THRESHOLD, "ignore_index": ignore_index, "num_groups": 2, "task": "all"},
+            groups=groups,
```

Comment:
By providing the groups tensor as a ``kwarg`` I am afraid that it will not always be on the right device, but accounting for this seems to require some big changes. Therefore, I will mark this draft as ready for review to get some feedback on this :).

Reply:
Update: I am unsure how to proceed with the testing as adding them as ``kwargs_update: Additional keyword arguments that will be passed with preds and target when running update on the metric`` seems to cause many problems.

Reply:
I agree. I think you could modify the baseclass tester here to use https://github.com/Lightning-AI/utilities/blob/main/src/lightning_utilities/core/apply_func.py#L23 on all kwargs to move tensors to the correct devices (e.g. of `preds`)